Sources/Extensions/CALayer+Installer.swift

```diff
@@ -0,0 +1,158 @@
+    /// - note: If you don't use a static width for instance, important to understand, that it's not correct to call 'left' and 'right' relations together by sublayers,
+    ///         because `container` sets width relatively width of sublayers and here is some ambiguous.
+    ///
+    /// - parameter container:                The layer where a container will be added.
+    /// - parameter relation:            The relation of `ContainerRelation` type.
+    ///     - `width`:                   The width of a container. If you specify a width only a dynamic height will be calculated.
+    ///     - `height`:                  The height of a container. If you specify a height only a dynamic width will be calculated.
+    ///     - `horizontal(left, right)`: The left and right insets of a container. If you specify these parameters only a dynamic height will be calculated.
+    ///     - `vertical(top, bottom)`:   The top and bottom insets of a container. If you specify these parameters only a dynamic width will be calculated.
+    /// - parameter installerBlock:      The installer block within which you should configure frames for all sublayers.
+
+    func configure(container: CALayer, relation: ContainerRelation? = nil, installerBlock: () -> Void) {
+        container.frame = .zero
+
+        var relationWidth: CGFloat?
+        var relationHeight: CGFloat?
+
+        if let relation = relation {
+            switch relation {
+            case let .width(width):
+                container.frame.size.width = width.value
+                relationWidth = width.value
+
+            case let .height(height):
+                container.frame.size.height = height.value
+                relationHeight = height.value
+
+            case let .horizontal(lInset, rInset):
+                container.configureFrame { maker in
+                    maker.left(inset: lInset).right(inset: rInset)
+                }
+                let width = container.frame.width
+                container.frame = .zero
+                container.frame.size.width = width
+                relationWidth = width
+
+            case let .vertical(tInset, bInset):
+                container.configureFrame { maker in
+                    maker.top(inset: tInset).bottom(inset: bInset)
+                }
+                let height = container.frame.height
+                container.frame = .zero
+                container.frame.size.height = height
+                relationHeight = height
+            }
+        }
+
+        for sublayer in self where sublayer.superlayer != container {
+            container.addSublayer(sublayer)
+        }
+
+        installerBlock()
+        container.configureFrame { maker in
+            maker._container()
+        }
+
+        if let width = relationWidth {
+            container.frame.size.width = width
+        }
+
+        if let height = relationHeight {
+            container.frame.size.height = height
+        }
+
+        installerBlock()
```

Comment:
Why installerBlack is called twice?

Reply:
Without it container and its content appears properly only on the next layout

Reply:
legacy code, nested from UIView+Installer